WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java

```diff
@@ -174,10 +176,14 @@ public void onBindViewHolder(GridViewHolder holder, int position) {
             } else {
                 holder.imageView.setImageUrl(getBestImageUrl(media), WPNetworkImageView.ImageType.PHOTO);
             }
-        } else if (media.isVideo() && !TextUtils.isEmpty(media.getThumbnailUrl())) {
+        } else if (media.isVideo()) {
             holder.fileContainer.setVisibility(View.GONE);
             holder.videoOverlayContainer.setVisibility(View.VISIBLE);
-            holder.imageView.setImageUrl(media.getThumbnailUrl(), WPNetworkImageView.ImageType.VIDEO);
+            if (isLocalFile) {
+                loadLocalVideoThumbnail(media.getFilePath(), holder.imageView);
+            } else {
+                holder.imageView.setImageUrl(media.getThumbnailUrl(), WPNetworkImageView.ImageType.VIDEO);
```

Comment:
`media.getThumbnailUrl()` seems to return `null` for self hosted sites. `ImageUtils.getVideoFrameFromVideo` should load the frame from the network instead. Note that's a blocking routine.